core/src/test/java/org/infinispan/remoting/transport/InitialClusterSizeTest.java

```diff
@@ -47,22 +53,29 @@ public void testInitialClusterSize() throws ExecutionException, InterruptedExcep
       assertEquals(CLUSTER_SIZE, manager(0).getMembers().size());
    }
 
-   @Test(expectedExceptions = CacheException.class, expectedExceptionsMessageRegExp = "ISPN000399:.*")
-   public void testInitialClusterSizeFail() throws Throwable {
-      try {
-         Future<?>[] threads = new Future[CLUSTER_SIZE - 1];
-         for (int i = 0; i < CLUSTER_SIZE - 1; i++) {
-            final int index = i;
-            threads[i] = fork(() -> {
-               manager(index).start();
-               return true;
-            });
-         }
-         for (Future<?> f : threads) {
-            f.get(CLUSTER_TIMEOUT_SECONDS + 1, TimeUnit.SECONDS);
+   public <T extends FieldElement<T>> void testInitialClusterSizeFail() throws Throwable {
+      List<Future<Void>> futures = new ArrayList<>();
+      for (int i = 0; i < CLUSTER_SIZE - 1; i++) {
+         EmbeddedCacheManager manager = manager(i);
+         futures.add(fork(() -> {
+            manager.start();
+            return null;
+         }));
+      }
+
+      for (Future<Void> future : futures) {
+         try {
+            // JGroupsTransport only starts counting down on initialClusterTimeout *after* it connects.
+            // The initial connection may take take 3 seconds (GMS.join_timeout) because of JGRP-2028
+            // Shutdown may also take 2 seconds (GMS.view_ack_collection_timeout) because of JGRP-2030
+            future.get(CLUSTER_TIMEOUT_SECONDS + 5, TimeUnit.SECONDS);
+            fail("Should have thrown an exception");
+         } catch (ExecutionException ee) {
+            Exceptions.assertException(EmbeddedCacheManagerStartupException.class, ee.getCause());
+            Exceptions.assertException(CacheException.class,
+                  org.infinispan.util.concurrent.TimeoutException.class, "ISPN000399:.*",
```

Comment:
Probably because of a long terms maintenance and different release cycles. I've been fixing those issues in the past, so I would advice being cautious.